hazelcast/src/main/java/com/hazelcast/internal/nio/tcp/TcpIpEndpointManager.java

```diff
@@ -176,7 +178,16 @@ public void addConnectionListener(ConnectionListener listener) {
 
     @Override
     public synchronized void accept(Packet packet) {
-        bindHandler.process(packet);
+        // the packet was sent from 3.12 if the 4_0 flag is missing
+        // 4.0.1 and 4.2 members do not set this flag
```

Comment:
Does 4.1 set it?

Reply:
Whoops, good catch. I'll send a PR for master.

Reply:
Fixed with https://github.com/hazelcast/hazelcast/pull/17756